[14.0][FIX] hr_expense: Amend multicurrency

victoralmau opened the pull request:
Prevent `company_currency_id` field from not having the correct value in some cases.

Steps to reproduce the error:

- Enable multi-currency.
- Go to Expenses > Expenses report and create a new one.
- Add a line (creating a new expense)
- Add another line (creating a new expense)
- Modify the unit price of a line
- Save the expense report
- The line that we have not modified the price, will not have the `company_currency_id` value set (the value of `total_amount_company` will also be incorrect).

**Before**
![ejemplo-antes](https://github.com/OCA/OCB/assets/4117568/0d5aaf03-f58b-4948-ab54-0c8958736a07)

**After**
![ejemplo-despues](https://github.com/OCA/OCB/assets/4117568/62d04dce-1499-4ec7-861f-4d0d85dfe1d8)

Please @pedrobaeza can you review it?

This change is NOT required in higher versions.

@Tecnativa TT46648

pedrobaeza reviewed (changes requested):
Please adopt the solution from v15 and v16 (indicating so in the commit message), and avoid changes in whitespaces of other parts.

victoralmau committed:
[FIX] hr_expense: Amend multicurrency

Change related of company_currency_id field
This change is part of what has been done at https://github.com/odoo/odoo/commit/2de15d5c0cd263475a5bee54cfdefc61972a36e8 to avoid incorrect values.

TT46648

victoralmau commented:
Changes done.
